src/legacy/core_plugins/timelion/public/panels/timechart/schema.ts

```diff
@@ -17,42 +17,51 @@
  * under the License.
  */
 
-require('./flot');
+import './flot';
 import _ from 'lodash';
 import $ from 'jquery';
 import moment from 'moment-timezone';
+import { timefilter } from 'ui/timefilter';
+import { npSetup } from 'ui/new_platform';
+// @ts-ignore
 import observeResize from '../../lib/observe_resize';
+// @ts-ignore
 import { calculateInterval, DEFAULT_TIME_FORMAT } from '../../../common/lib';
-import { timefilter } from 'ui/timefilter';
+import { TimelionStartDependencies } from '../../plugin';
+import { tickFormatters } from '../../services/tick_formatters';
+import { xaxisFormatterProvider } from './xaxis_formatter';
+import { generateTicksProvider } from './tick_generator';
 
 const DEBOUNCE_DELAY = 50;
+const { uiSettings } = npSetup.core;
```

Comment:
it should be passed from dependencies. See Vega, TSVB, Metrics plugin as example

Reply:
Moved